### PR TITLE
cake 5: Sort param and return type hints

### PR DESCRIPTION
**Pull request by othercorey**
Refs https://github.com/cakephp/cakephp/issues/15528

### CODE REVIEW

File: CakePHP/Sniffs/Commenting/FunctionCommentSniff.php [hunk 1]
```diff
@@ -134,6 +148,158 @@ protected function processTagSpacing(File $phpcsFile, int $stackPtr, int $commen
...
+            'iterable',
+            'array',
+            'string',
+            'float',
+            'int',
+            'bool',
+            'true',
+            'false',
+            'null',
+            'void',
```
Comment (ADmad) [inline CakePHP/Sniffs/Commenting/FunctionCommentSniff.php:252]: Is void required in this list given that it can only be used by itself?

Comment (othercorey) [reply CakePHP/Sniffs/Commenting/FunctionCommentSniff.php:252]: `void` is allowed in union types such as the return type of actions. The default, undeclared return type is `mixed|void` according to php.

Comment (dereuromark) [reply CakePHP/Sniffs/Commenting/FunctionCommentSniff.php:252]: Makes no sense, should have been null ;) But meeh.

Comment (ADmad) [reply CakePHP/Sniffs/Commenting/FunctionCommentSniff.php:252]: > void is allowed in union types such as the return type of actions.

You mean in docblock types right? Surely you can't have `void` in a union in concreate PHP type.

Comment (ADmad) [reply CakePHP/Sniffs/Commenting/FunctionCommentSniff.php:252]: > Makes no sense, should have been null ;) But meeh.

I think the `void` in docblocks type is to make static analyzers happy else they will whine about missing return.

**Comment by othercorey** [comment]
I moved this into a separate sniff file and extended it to support var tags. Also extended to support psalm- and phpstan- pefixes.

**Comment by othercorey** [comment]
Updated to include `type[]` to `array<type>` type hint conversion.

**Pull request merged**